### PR TITLE
[BUGFIX] Adapt GoTo plugin for oauth2 authentication

### PR DESCRIPTION
**Pull request by FlorianWessels**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N 
| Automated tests included? | Y
| Related user documentation PR URL | N
| Related developer documentation PR URL | N
| Issues addressed (#s or URLs) | #5736
| BC breaks? | N
| Deprecations? | N 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Migrate GoTo authentication to new RFC 6749 compliant OAuth Proxy.

### CODE REVIEW

**Comment by npracht** [comment]
@FlorianWessels Can you check what's wrong with travis please ?

**Comment by FlorianWessels** [comment]
Yeah, I will do that.

**Comment by npracht** [comment]
@johbuch can you confirm that with this new version of the GoTo API, everything is working properly ?

**Comment by johbuch** [comment]
I tested to set it up on mautibox. I can confirm that i could authorised the app without any problem.
Then I could find new filters in segments but I could not properly test the sync because we need a specific cronjob to run.
But before (without this PR), we could not set up the sync because of the Oauth1. But this seems to be fixed

**Comment by kuzmany** [comment]
Code looks good.
What about backward compatibility? Works after upgrade?

**Comment by npracht** [comment]
Yes you just need to re-auth.
But without it, it won't work anymore....

**Comment by kuzmany** [comment]
Now I got it from issue https://github.com/mautic/mautic/issues/5736
Better in future write that to description. Code it's legit.
Do you use it in production?

**Comment by npracht** [comment]
Yes we do !

File: plugins/MauticCitrixBundle/EventListener/IntegrationRequestSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * @param array $parameters
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    protected function getAuthorization(array $parameters)
+    {
+        if (!isset($parameters['client_id']) || empty($parameters['client_id'])) {
+            throw new \Exception('No client ID given.', 1554211764);
```
Comment (anton-vlasenko) [inline plugins/MauticCitrixBundle/EventListener/IntegrationRequestSubscriber.php]: @FlorianWessels 
Excuse me, but what is the purpose of the exception code? I mean `1554211764`.
Is it used somewhere? It isn't mentioned in the API docs.

File: plugins/MauticCitrixBundle/EventListener/IntegrationRequestSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        if (!isset($parameters['client_secret']) || empty($parameters['client_secret'])) {
+            throw new \Exception('No client secret given.', 1554211808);
```
Comment (anton-vlasenko) [inline plugins/MauticCitrixBundle/EventListener/IntegrationRequestSubscriber.php]: Same here.

**Pull request merged**